crates/rpc-client/src/poller.rs

```diff
@@ -15,14 +16,22 @@ use tracing::Instrument;
 /// The number of retries for polling a request.
 const MAX_RETRIES: usize = 3;
 
-/// A Poller task.
+/// A poller task builder.
+///
+/// This builder is used to create a poller task that repeatedly polls a method on a client and
+/// sends the responses to a channel. By default, this is done every 10 seconds, with a channel size
+/// of 16, and no limit on the number of successful polls. This is all configurable.
+///
+/// The builder is consumed using the [`spawn`](Self::spawn) method, which returns a channel to
+/// receive the responses. The task will continue to poll until either the client or the channel is
+/// dropped.
+///
+/// The channel can be converted into a stream using the [`into_stream`](PollChannel::into_stream)
+/// method.
+// TODO: make this be able to be spawned on the current thread instead of forcing a task.
```

Comment:
why does that matter

Reply:
this crate depends on tokio only for this spawn+channel, could be an option also to do everything in the current thread